### DISCUSSION TITLE
osu!taiko Editor Problems/Suggestions

### DISCUSSION THREAD

**Discussion by zachmanthethird** [ppy/osu#14398] [Q&A]
I've decided to have a gander at the new osu!taiko editor and try my hand at mapping. I do not have any prior mapping experience, nor are these points intended to be biased.
1. Upon entering the editor via the Main Menu and selecting an audio track on the setup screen, some interesting errors occur. Kiai timing points don't connect on the bottom timeline. Difficulty (in this context slider velocity) timing points have no effect; additionally, the default is lower than it should be.
![osu_2021-08-19_09-36-37](https://user-images.githubusercontent.com/74262407/130112265-b2e89898-2507-4745-9d66-fe2acd89b476.png)
![osu_2021-08-19_09-36-55](https://user-images.githubusercontent.com/74262407/130112306-0a9eed37-fde9-46f3-a5ea-c60b2b1a3196.png)
When saving and reloading the map from the song selection screen, these errors disappear: ![osu_2021-08-19_09-43-24](https://user-images.githubusercontent.com/74262407/130112471-8c5950ec-edd6-4ecb-8770-2e702997c95c.png)
2. There are two visual glitches when maneuvering the editor. The first is how the drum freezes when moving significantly around the timeline (see previous screenshots). This can cause inaccurate representations of what is being played. The second glitch is that some notes render without their center symbols. I've replicated this effect many times, but never in the same way. However, they fix themselves upon being hit, making this a minor problem at best.
![osu_2021-08-19_09-42-47](https://user-images.githubusercontent.com/74262407/130113544-ac33fa40-decd-4328-92a4-f04df2274670.png)
3. I assume #14323 will fix this next problem, but I can't be 100% sure. It is possible to place notes before the starting point of the mp3 as long as your cursor is still within the playfield.
![osu_2021-08-19_10-17-06](https://user-images.githubusercontent.com/74262407/130114363-b789e362-ef84-4420-9fc4-50e6cab28e11.png)
The verification system doesn't detect hit sounds even after removing the preceding notes. I haven't found anything that removes this.
![osu_2021-08-19_10-14-50](https://user-images.githubusercontent.com/74262407/130115091-6f1595b3-07ae-4297-b5d6-34e8eb4bac9e.png)
![osu_2021-08-19_10-14-56](https://user-images.githubusercontent.com/74262407/130115100-59e30920-8ddc-4f27-9ff2-3b4df54c81a3.png)
4. After attempting to replicate the previous issue, I found another one. Creating a timing point before selecting an audio track causes the editor to behave in strange ways. Again, reloading the map removes the problem.
![osu_2021-08-19_09-49-08](https://user-images.githubusercontent.com/74262407/130115679-7355a1d4-f046-4a65-95db-7b5713e3edd8.png)
5. The last suggestion I have _is_ my personal opinion. It took me a moment to figure out that left-click was for dons and right-click was for kats. However, if I select whistle/clap for my hit sound, should a left-click not create a kat like a finish hit sound creates a finisher? This seems intuitively obvious; however, I am unaware if stable acts the same way. I understand that this is a design preference and should be treated as such, though this seems like an easy change from a layman's perspective.

While I might usually turn most of these into issues, I want to ensure they're replicable on other lazer clients before escalating them further, especially after #14336 proved not to be an actual problem. Otherwise, I thoroughly enjoy this new editor and greatly anticipate how storyboarding will be implemented. :smiley:

**Comment by peppy** [comment]
`3` is likely a misunderstanding. it's not asking for a sample change but hitsounds applied to individual objects.

**Reply by zachmanthethird** [reply]
Thank you for clarifying. It doesn't appear that having notes before the mp3 starts (which is still possible) has any influence. However, it doesn't change status on the verify screen until after saving and loading.

Here is an example of hitsounds present:
![osu_2021-08-20_14-58-33](https://user-images.githubusercontent.com/74262407/130297722-e4ae95a0-95a8-460a-89ec-dbb39a582704.png)
![osu_2021-08-20_14-58-38](https://user-images.githubusercontent.com/74262407/130297727-d1278bd1-bb9a-420c-8348-f08db1ae8ec0.png)

And here is an example of hitsounds not present:
![osu_2021-08-20_14-59-38](https://user-images.githubusercontent.com/74262407/130297732-0d28d3dc-0fb5-4a73-927d-167b2881ff6f.png)
![osu_2021-08-20_14-59-42](https://user-images.githubusercontent.com/74262407/130297736-df528fa9-d23f-4eb8-8c46-edc426f0b13d.png)

**Reply by peppy** [reply]
Nowhere in those screenshots are you showing any hitsounds present. Unless you mean don/kat?
